packages/assets-controllers/src/DeFiPositionsController/DeFiPositionsController.ts

```diff
@@ -0,0 +1,247 @@
+import {
+  groupDeFiPositions,
+  type GroupedDeFiPositions,
+} from './group-defi-positions';
+import { reduceInBatchesSerially } from '../assetsUtil';
+
+const TEN_MINUTES_IN_MS = 60_000;
+
+const FETCH_POSITIONS_BATCH_SIZE = 10;
+
+const controllerName = 'DeFiPositionsController';
+
+type GroupedDeFiPositionsPerChain = {
+  [chain: Hex]: GroupedDeFiPositions;
+};
+
+export type DeFiPositionsControllerState = {
+  /**
+   * Object containing DeFi positions per account and network
+   */
+  allDeFiPositions: {
+    [accountAddress: string]: GroupedDeFiPositionsPerChain | null;
+  };
+};
+
+const controllerMetadata: StateMetadata<DeFiPositionsControllerState> = {
+  allDeFiPositions: {
+    persist: false,
+    anonymous: false,
+  },
+};
+
+export const getDefaultDefiPositionsControllerState =
+  (): DeFiPositionsControllerState => {
+    return {
+      allDeFiPositions: {},
+    };
+  };
+
+export type DeFiPositionsControllerActions =
+  DeFiPositionsControllerGetStateAction;
+
+export type DeFiPositionsControllerGetStateAction = ControllerGetStateAction<
+  typeof controllerName,
+  DeFiPositionsControllerState
+>;
+
+export type DeFiPositionsControllerEvents =
+  DeFiPositionsControllerStateChangeEvent;
+
+export type DeFiPositionsControllerStateChangeEvent =
+  ControllerStateChangeEvent<
+    typeof controllerName,
+    DeFiPositionsControllerState
+  >;
+
+/**
+ * The external actions available to the {@link DeFiPositionsController}.
+ */
+export type DeFiPositionsControllerAllowedActions =
+  AccountsControllerListAccountsAction;
+
+/**
+ * The external events available to the {@link DeFiPositionsController}.
+ */
+export type DeFiPositionsControllerAllowedEvents =
+  | KeyringControllerUnlockEvent
+  | KeyringControllerLockEvent
+  | TransactionControllerTransactionConfirmedEvent
+  | AccountsControllerAccountAddedEvent;
+
+/**
+ * The messenger of the {@link DeFiPositionsController}.
+ */
+export type DeFiPositionsControllerMessenger = RestrictedMessenger<
+  typeof controllerName,
+  DeFiPositionsControllerActions | DeFiPositionsControllerAllowedActions,
+  DeFiPositionsControllerEvents | DeFiPositionsControllerAllowedEvents,
+  DeFiPositionsControllerAllowedActions['type'],
+  DeFiPositionsControllerAllowedEvents['type']
+>;
+
+/**
+ * Controller that stores assets and exposes convenience methods
+ */
+export class DeFiPositionsController extends StaticIntervalPollingController()<
+  typeof controllerName,
+  DeFiPositionsControllerState,
+  DeFiPositionsControllerMessenger
+> {
+  readonly #fetchPositions: (
+    accountAddress: string,
+  ) => Promise<DefiPositionResponse[]>;
+
+  readonly #isEnabled: () => boolean;
+
+  /**
+   * DeFiPositionsController constuctor
+   *
+   * @param options - Constructor options.
+   * @param options.messenger - The controller messenger.
+   * @param options.isEnabled - Function that returns whether the controller is enabled. (default: () => true)
+   */
+  constructor({
+    messenger,
```

Comment:
Can you add a `state` argument to this constructor and then mix it in with the default state when calling `super`? This will align with our standards around controllers.

You can read here for more: https://github.com/MetaMask/core/blob/main/docs/controller-guidelines.md#accept-an-optional-partial-representation-of-state

Reply:
This controller does not have persisted state.

If I add a state, then I have to add tests to verify that state is being added, even though it is not needed nor wanted as we would never pass state when initialising the controller.

Reply:
Ah, I see, I missed that, thanks for pointing that out. What you have makes sense then.